pkg/clients/helm/client.go

```diff
@@ -192,6 +203,20 @@ func (hc *client) pullChart(spec *v1beta1.ChartSpec, creds *RepoCreds, chartDir
 	pc.Password = creds.Password
 
 	pc.DestDir = chartDir
+	
+	if registry.IsOCI(spec.Repository) {
+		parsedURL, err := url.Parse(spec.Repository)
+		if err != nil {
+			return errors.Wrap(err, errFailedToParseURL)
+		}
+		var out strings.Builder
+		// TODO add support for insecure
```

Comment:
Where is a good place to add support for configuring insecure registries? I suppose we can add it as a property in the Release manifest but I would appreciate some suggestions for naming that property.

Reply:
Hmm, good question. 
I would say somewhere under `Release` make sense like `spec.forProvider.chart.insecure`. But this would indicate that `Chart is insecure` but indeed repo/registry is insecure. 

So maybe something like

```
spec:
# rollbackLimit: 3
  forProvider:
    chart:
      name: wordpress
      repository: https://charts.bitnami.com/bitnami
      version: 9.3.19
      # Deprecated use pull.SecretRef instead
      # pullSecretRef:
      pull:
        insecure: true
        SecretRef:
          name: museum-creds
          namespace: default
```

But then we may also want to have repository and URL under pull as well 🤔

Reply:
We could also add it to the [Args struct](https://github.com/crossplane-contrib/provider-helm/blob/master/pkg/clients/helm/args.go ):

e.g the helm flag `--insecure-skip-tls-verify` as `InsecureSkipTLSVerify` to also add support for pulling charts from insecure repositories that are non oci registries. By that we can use the same flag to pass to the registry login helm command to indicate that the login is insecure.

```yaml
spec:
# rollbackLimit: 3
  forProvider:
    chart:
      name: wordpress
      repository: https://charts.bitnami.com/bitnami
      version: 9.3.19
    insecureSkipTLSVerify: true
    pullSecretRef:
      name: museum-creds
      namespace: default
```

Reply:
Yeah, that sounds like a better idea considering all those args are optional `flags` for `helm install` command.